File: backend/src/main/java/org/akashihi/mdg/dao/TransactionRepositoryCustom.java
package org.akashihi.mdg.dao;

import org.akashihi.mdg.entity.Transaction;
import org.springframework.data.jpa.domain.Specification;

import java.util.stream.Stream;

public interface TransactionRepositoryCustom {
    public Stream<Transaction> streamByAccount(Specification<Transaction> spec);
}

File: backend/src/main/kotlin/org/akashihi/mdg/dao/TransactionRepositoryCustom.kt
package org.akashihi.mdg.dao

import org.akashihi.mdg.entity.Transaction
import org.springframework.data.jpa.domain.Specification
import java.util.stream.Stream

interface TransactionRepositoryCustom {
    fun streamByAccount(spec: Specification<Transaction>): Stream<Transaction>
}